client/src/pages/RoleApproval.js

```diff
@@ -12,24 +12,28 @@ import {
   Label,
   Input,
 } from 'reactstrap';
+import Loader from 'react-loader-spinner';
 
 import { getUsersForRolesPage, changeRole } from '../utils/auth';
 
 const RoleApproval = () => {
+  const [loading, setLoading] = useState(false);
   const [newRole, setNewRole] = useState('');
   const [response, setResponse] = useState('');
   const [password, setPassword] = useState('');
   const [users, setUsers] = useState([]);
   const [userWithNewRole, setUserWithNewRole] = useState(-1);
 
   useEffect(() => {
+    setLoading(true);
     async function fetchData() {
       const userData = await getUsersForRolesPage();
       const userDataParsed = await userData.json();
 
       if (userDataParsed.user_emails) {
         setUsers(userDataParsed.user_emails);
       }
+      setLoading(false);
     }
     fetchData();
   }, [setUsers]);
```

Comment:
I feel like this would make more sense to `setLoading(true)` once the fetch data function begins, not outside of it

Reply:
done